msdsn/conn_str_test.go

```diff
@@ -217,6 +217,25 @@ func TestConnParseRoundTripFixed(t *testing.T) {
 	}
 }
 
+func TestServerNameInTLSConfig(t *testing.T) {
+
+	var tests = []struct {
+		dsn  string
+		host string
+	}{
+		{"sqlserver://someuser:somepass@somehost?TrustServerCertificate=false&encrypt=true", "somehost"},
+		{"sqlserver://someuser:somepass@somehost?TrustServerCertificate=false&encrypt=true&hostnameincertificate=someotherhost", "someotherhost"},
```

Comment:
>true& [](http://example.com/codeflow?start=80&length=5)

if encrypt isn't set, what should cfg.Host be? is using `encrypt=DISABLED&hostnameincertificate=something` 
or
`encrypt=false&hostnameincertificate=something` 

undefined options?

Reply:
hmm, tricky. I added tests that restore the previous logic. This means:

```
encrypt = DISABLED -> tls.Config is nil
encrypt = false -> empty tls.Config, but with hostname set
```

I'm not sure what the right logic would be, but in case of doubt, I would stick to the previous logic to stay backwards compatible.

Reply:
alternatively we could make it an error to supply `hostnameincertificate` when encrypt is not true.

Reply:
backward compatible is fine for now.
Once we branch a v2 we'll revisit it as part of the "TDS8" implementation.
Note that ADO.Net just ignores the property when Encrypt is not "Mandatory" or "Strict"
https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlconnectionstringbuilder.hostnameincertificate?view=sqlclient-dotnet-standard-5.1

Reply:
Yeah, probably a good idea to refactor this stuff for v2. I also saw a 2nd definition of the encrypt enum which should be merged into one.

Anyway, thanks for approving. As I lack write access, I guess you or one of the maintainers has to merge it.